### ISSUE TITLE
Wrong Content-MD5

### ISSUE THREAD

**Issue by sergei66666** [minio/minio-dotnet#435]
Hello)

In `V4Authenticator` in function `SetContentMd5` happen calculation of Content-MD5 header.

There is one case, which we should add before
```
if (body == null)
{
  body = new byte[0];
}
```

It is
```
if(bodyParameter.Value is XElement)
{
   var str = request.XmlSerializer.Serialize(bodyParameter.Value);
   body = System.Text.Encoding.UTF8.GetBytes(str);
}
```
 Without it such methods, for examle batch verion of `RemoveObjectAsync` throw error with message "bad md5".

My fix is bad from performance view (becase there is will be two conversions from XElement to string). To fix it I see only one way - change all methods which use XElement, But it is very dangerous.

**Comment by sergei66666** [comment]
But.. We can make it other way:

```
byte[] body = null; <--- after this line (568)

if (bodyParameter.Value is XElement)
{
  bodyParameter.DataFormat = DataFormat.None;
 bodyParameter.Value = request.XmlSerializer.Serialize(bodyParameter.Value);
}
```

**Comment by harshavardhana** [comment]
We are generally happy for a PR @sergei66666 - feel free to send it.

**Comment by sergei66666** [comment]
OK. I just saw a lot of unacceptable requests and did not do them personally. I will do it

**Comment by poornas** [comment]
this was fixed by #439
